dotnet/src/Agents/OpenAI/AssistantThreadActions.cs

```diff
@@ -0,0 +1,525 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure;
+using Azure.AI.OpenAI.Assistants;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.ChatCompletion;
+
+namespace Microsoft.SemanticKernel.Agents.OpenAI;
+
+/// <summary>
+/// Actions associated with an Open Assistant thread.
+/// </summary>
+internal static class AssistantThreadActions
```

Comment:
NOTE: This code has been ported from `OpenAIAssistantChannel` with no disruption to the core processing which has been stabilized over several cycles.